neqo-http3/tests/webtransport/mod.rs

```diff
@@ -0,0 +1,205 @@
+
+    let out = server.process(out.dgram(), now());
+    let out = client.process(out.dgram(), now());
+    let out = server.process(out.dgram(), now());
+    assert!(out.as_dgram_ref().is_none());
+
+    let authentication_needed = |e| matches!(e, Http3ClientEvent::AuthenticationNeeded);
+    assert!(client.events().any(authentication_needed));
+    client.authenticated(AuthenticationStatus::Ok, now());
+
+    let out = client.process(out.dgram(), now());
+    let connected = |e| matches!(e, Http3ClientEvent::StateChange(Http3State::Connected));
+    assert!(client.events().any(connected));
+
+    assert_eq!(client.state(), Http3State::Connected);
+
+    // Exchange H3 setttings
+    let out = server.process(out.dgram(), now());
+    let out = client.process(out.dgram(), now());
+    let out = server.process(out.dgram(), now());
+    let out = client.process(out.dgram(), now());
+    let _ = server.process(out.dgram(), now());
+}
+
+fn connect(wt_enable_client: bool, wt_enable_server: bool) -> (Http3Client, Http3Server) {
+    let mut client = default_http3_client(wt_enable_client);
+    let mut server = default_http3_server(wt_enable_server);
+    connect_with(&mut client, &mut server);
+    (client, server)
+}
+
+struct WtTest {
+    client: Http3Client,
+    server: Http3Server,
+}
+
+impl WtTest {
+    pub fn new() -> Self {
+        let (client, server) = connect(true, true);
+        Self { client, server }
+    }
+
+    fn negotiate_wt_session(&mut self, accept: bool) -> (u64, Option<WebTransportRequest>) {
+        let wt_session_id = self
+            .client
+            .webtransport_create_session(now(), &("https", "something.com", "/"))
+            .unwrap();
+        self.exchange_packets();
+
+        let mut wt_server_session = None;
+        while let Some(event) = self.server.next_event() {
+            match event {
+                Http3ServerEvent::WebTransport(
+                    WebTransportServerEvent::WebTransportNewSession {
+                        mut session,
+                        headers,
+                    },
+                ) => {
+                    assert!(
+                        headers
+                            .iter()
+                            .any(|h| h.name() == ":method" && h.value() == "CONNECT")
+                            && headers
+                                .iter()
+                                .any(|h| h.name() == ":protocol" && h.value() == "webtransport")
+                    );
+                    session.response(accept).unwrap();
+                    wt_server_session = Some(session);
+                }
+                Http3ServerEvent::Data { .. } => {
+                    panic!("There should not be ane data events!");
+                }
+                _ => {}
+            }
+        }
+
+        self.exchange_packets();
+        (wt_session_id, wt_server_session)
+    }
+
+    fn create_wt_session(&mut self) -> (u64, WebTransportRequest) {
```

Comment:
What is wrong with using `WebTransportRequest::stream_id()` here?  You should just be able to assert_eq!() to ensure that the two identifiers line up.